app/controllers/settings/profiles_controller.rb

```diff
@@ -16,11 +16,32 @@ def update
       render :edit, status: :unprocessable_entity
     end
   end
+  
+  def destroy
+    begin
+      delete_user
+      logout
+      redirect_to new_registration_path, notice: t(".success")
+    rescue ActiveRecord::RecordNotDestroyed => e
+      Rails.logger.error "Error deleting account: #{e.message}"
+      return redirect_to settings_profile_path, alert: t(".account_deletion_failed")
+    rescue StandardError => e
+      Rails.logger.error "An unexpected error occurred: #{e.message}"
+      redirect_to settings_profile_path, alert: t(".account_deletion_failed")
+    end
+  end
 
   private
 
   def user_params
     params.require(:user).permit(:first_name, :last_name,
                                  family_attributes: [ :name, :id ])
   end
+
+  def delete_user
+    attributes = Current.user.attributes
+    Current.user.destroy!
+
+    DeleteUserJob.perform_later(attributes)
```

Comment:
While I think you're right that we should be doing this in a background job, I think a potentially more stable strategy would be something like this:

```rb
def mark_user_for_deletion_and_logout
  Current.user.update!(marked_for_deletion: true)
  DeleteUserJob.perform_later(Current.user)
  logout
end
```

And then on login, we could check `marked_for_deletion` and fail early if it is marked `true`.

This would allow us to work directly with the `User` record all in one spot and avoid some of the error handling here.

Reply:
FYI - https://github.com/maybe-finance/maybe/issues/675#issuecomment-2081051730